File: 3254_find-the-power-of-k-size-subarrays_i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& a, int b) {
        int c = a.size(), d = 1;
        vector<int> e(c - b + 1, -1);
        for (int f = 1; f < b; f++) d = (a[f] == a[f - 1] + 1) ? d + 1 : 1;
        if (d == b) e[0] = a[b - 1];
        for (int g = 1, h = b; h < c; g++, h++) {
            d = (a[h] == a[h - 1] + 1) ? d + 1 : 1;
            if (d >= b) e[g] = a[h];
        }
        return e;
    }
};

File: 3254_find_the_power_of_k_size_subarrays_i..java
class Solution {
    public int[] resultsArray(int[] arr, int len) {
        int n = arr.length, cnt = 1;
        int[] out = new int[n - len + 1];
        Arrays.fill(out, -1);
        for (int i = 1; i < len; i++) 
            cnt = (arr[i] == arr[i - 1] + 1) ? cnt + 1 : 1;
        if (cnt == len) out[0] = arr[len - 1];
        for (int left = 1, right = len; right < n; left++, right++) {
            cnt = (arr[right] == arr[right - 1] + 1) ? cnt + 1 : 1;
            if (cnt >= len) out[left] = arr[right];
        }
        return out;
    }
}
